sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala

```diff
@@ -76,7 +85,34 @@ class LogicalPlanToSQLSuite extends SQLBuilderTest with SQLTestUtils {
     }
   }
 
-  private def checkHiveQl(hiveQl: String): Unit = {
+  // Used for generating new query answer files by saving
+  private val saveQuery = false
+
+  /**
+   * Compare the generated SQL with the expected answer string.
+   * Note that there exists a normalization for both arguments for the convenience.
+   * - Remove the id from the generated attributes, e.g., `gen_attr_1` -> `gen_attr`.
+   */
+  private def checkSQLStructure(originalSQL: String, convertedSQL: String, answerFile: String) = {
+    val normalizedGenSQL = convertedSQL.replaceAll("`gen_attr_\\d+`", "`gen_attr`")
+    if (answerFile != null) {
+      val path = s"src/test/resources/sqlgen/$answerFile.sql"
+      val separator = "-" * 80
+      if (saveQuery) {
+        val answerText = s"${originalSQL.trim()}\n${separator}\n$normalizedGenSQL"
+        Files.write(Paths.get(path), answerText.getBytes(StandardCharsets.UTF_8))
+      } else {
+        val answerText = new String(Files.readAllBytes(Paths.get(path)), StandardCharsets.UTF_8)
+        val sqls = answerText.split(separator)
+        if (sqls.length == 2) {
+          val normalizedExpectSQL = sqls(1).trim()
+          assert(normalizedGenSQL == normalizedExpectSQL)
+        }
+      }
+    }
+  }
+
+  private def checkHiveQl(hiveQl: String, answerFile: String = null): Unit = {
```

Comment:
while you at it, rename checkHiveQl to something else (e.g. checkSQL).

Also why should answerFile be null?